### DISCUSSION TITLE
OIDC call to /token missing client_id/client_secret fields

### DISCUSSION THREAD

**Discussion by berkutta** [pocketbase/pocketbase#5495] [Q&A] [answered]
When connecting PocketBase to a custom OIDC server, I noticed that it doesn't send the `client_id` and `client_secret` fields to the `/token` endpoint.

Any idea how to debug why this is happening? I suspect there might be a bug/behavior in the custom OIDC server that's causing PocketBase to omit these fields.

Here’s the PocketBase call to the `/token` endpoint:
```
/token
{
  code: 'XXX',
  grant_type: 'authorization_code',
  redirect_uri: 'http://127.0.0.1:8090/redirect.html'
}
```

And here’s a call from another OIDC client to the same `/token` endpoint:
```
/token
{
  grant_type: 'authorization_code',
  redirect_uri: 'http://127.0.0.1:3001/auth/oidc.callback',
  client_id: 'your-client-id',
  client_secret: 'your-secret-id',
  code: 'XXX'
}
```

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not able to reproduce it and it works correctly for me when testing locally with Authentik.

Are you sure that you are logging the correct data? Is your custom OIDC server configured to expects the client id/secret to be as params or in the headers?

We use the std Go [x/oauth2](https://pkg.go.dev/golang.org/x/oauth2) package which technically should works with both styles but it expects the server to return proper error response in order for the auth style probing to work (see https://cs.opensource.google/go/x/oauth2/+/master:internal/token.go;l=212-253).

**Reply by berkutta** [reply]
Thanks for pointing out the header part!
This was the issue. [x/oauth2](https://pkg.go.dev/golang.org/x/oauth2) tries to send client_id/client_secret as auth header first, and if that fails, it sends these values as parameters.
My custom OIDC Server didn't respond with 401, but also didn't use the header fields. Changing it to respond with 401 when there are no client_id/client_secret fields results in PocketBase sending these values.

This behavior is documented here:
https://cs.opensource.google/go/x/oauth2/+/master:oauth2.go;l=93-95?q=%2F%2F%20AuthStyleAutoDetect%20means%20to%20auto-detect%20which%20authentication&ss=go%2Fx%2Foauth2
